docs/integrations/python.md

```diff
@@ -432,6 +434,101 @@ True
 
 For the documentation of lakeFS’s Python package and full api reference, see [https://pydocs-lakefs.lakefs.io](https://pydocs-lakefs.lakefs.io)
 
+## Using lakefs-spec for higher-level file operations
+
+The [lakefs-spec](https://lakefs-spec.org/latest/) project
+provides higher-level file operations on lakeFS objects with a filesystem API,
+built on the [fsspec](https://github.com/fsspec/filesystem_spec) project.
+
+**Note** This library is a third-party package and not maintained by the lakeFS developers; please file issues and bug reports directly
+in the [lakefs-spec](https://github.com/aai-institute/lakefs-spec) repository.
+ {: .note}
+
+### Installation
+
+Install `lakefs-spec` directly with `pip`:
+
+```
+python -m pip install --upgrade lakefs-spec
+```
+
+### Interacting with lakeFS through a file system
+
+To write a file directly to a branch in a lakeFS repository, consider the following example:
+
+```python
+from pathlib import Path
+
+from lakefs_spec import LakeFSFileSystem
+
+REPO, BRANCH = "example-repo", "main"
+
+# Prepare a local example file.
+lpath = Path("demo.txt")
+lpath.write_text("Hello, lakeFS!")
+
+fs = LakeFSFileSystem()  # will auto-discover credentials from ~/.lakectl.yaml
+rpath = f"{REPO}/{BRANCH}/{lpath.name}"
+fs.put(lpath, rpath)
+```
+
+Reading it again from remote is as easy as the following:
+
+```python
+f = fs.open(rpath, "rt")
+print(f.readline())  # prints "Hello, lakeFS!"
+```
+
+Many more operations like retrieving an object's metadata or checking an
+object's existence on the lakeFS server are also supported. For a full list,
+see the [API reference](https://lakefs-spec.org/latest/reference/lakefs_spec/).
+
+### Using transactions for atomic versioning operations
+
+As with the high-level SDK (see above), lakefs-spec also supports transactions
+for conducting versioning operations on newly modified files. The following is an example of creating a commit on the repository's main branch directly after a file upload:
+
+```python
+from lakefs_spec import LakeFSFileSystem
+
+fs = LakeFSFileSystem()
+
+# assumes you have a local train-test split as two text files:
+# train-data.txt, and test-data.txt.
+with fs.transaction("example-repo", "main") as tx:
+    fs.put_file("train-data.txt", f"example-repo/{tx.branch.id}/train-data.txt")
+    tx.commit(message="Add training data")
+    fs.put_file("test-data.txt", f"example-repo/{tx.branch.id}/test-data.txt")
+    sha = tx.commit(message="Add test data")
+    tx.tag(sha, name="My train-test split")
+```
+
+Transactions are atomic - if an exception happens at any point of the transaction, the repository remains unchanged.
+
+### Integrations with popular data science packages
```

Comment:
Might bring this even higher (before transactions) as I believe it'll be directly applicable to a larger audience

Reply:
Moved up above the transaction bit, just after the read/write.